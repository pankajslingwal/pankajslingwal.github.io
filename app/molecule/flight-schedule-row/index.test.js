import React from 'react';
import renderer from 'react-test-renderer';
import FlightScheduleTableRow from './';

describe("Flight Schedule Row", () => {

    it('renders correctly, no data passed', () => {
        const tree = renderer.create(<FlightScheduleTableRow />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly, data passed, selected is true', () => {
        const props = {
            key:"Emirates (EK512)",
            rowid:"Emirates (EK512)",
            flight:"Emirates (EK512)",
            from:"DXB",
            to:"BOM",
            dtime:"12:10",
            atime:"13:15",
            acode:"EK",
            duration:"5 hours 30 minutes",
            onRowClicked:jest.fn(),
            selected:true
          };

        const tree = renderer.create(<FlightScheduleTableRow {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly, data passed, selected is false', () => {
        const props = {
            key:"Emirates (EK512)",
            rowid:"Emirates (EK512)",
            flight:"Emirates (EK512)",
            from:"DXB",
            to:"BOM",
            dtime:"12:10",
            atime:"13:15",
            acode:"EK",
            duration:"5 hours 30 minutes",
            onRowClicked:jest.fn(),
            selected:false
          };

        const tree = renderer.create(<FlightScheduleTableRow {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});