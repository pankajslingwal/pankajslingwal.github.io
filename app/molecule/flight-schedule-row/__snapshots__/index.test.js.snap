// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight Schedule Row renders correctly, data passed, selected is false 1`] = `
<ul
  aria-label="Flight Emirates (EK512) departs from DXB at 12:10 and arrives BOM at 13:15, total Flight duration is 5 hours 30 minutes."
  className="schedule-row "
  role="button"
  tabIndex="0"
>
  <li
    className="only-desktop"
  >
    <span
      className="highlight"
    >
      12:10
    </span>
     (
    DXB
    )
  </li>
  <li
    className="only-desktop"
  >
    <span
      className="highlight"
    >
      13:15
    </span>
     (
    BOM
    )
  </li>
  <li
    className="only-mobile"
  >
    <span
      className="highlight"
    >
      12:10
       - 
      13:15
    </span>
  </li>
  <li
    className="only-mobile"
  >
    <span>
      DXB
       - 
      BOM
    </span>
  </li>
  <li
    className="only-mobile"
  >
    5 hours 30 minutes
  </li>
  <li>
    <span
      className="color-highlight icon "
    >
      <img
        alt="Flight Emirates (EK512) logo"
        src="test-file-stub"
      />
      Emirates (EK512)
    </span>
  </li>
  <li
    className="only-desktop"
  >
    5 hours 30 minutes
  </li>
  <li>
    <button
      aria-label="Select Flight number Emirates (EK512)"
      aria-pressed={false}
      className=""
      onClick={[Function]}
      title="Select Flight number Emirates (EK512)"
    >
      Select Flight
    </button>
  </li>
</ul>
`;

exports[`Flight Schedule Row renders correctly, data passed, selected is true 1`] = `
<ul
  aria-label="Flight Emirates (EK512) departs from DXB at 12:10 and arrives BOM at 13:15, total Flight duration is 5 hours 30 minutes."
  className="schedule-row selected-row"
  role="button"
  tabIndex="0"
>
  <li
    className="only-desktop"
  >
    <span
      className="highlight"
    >
      12:10
    </span>
     (
    DXB
    )
  </li>
  <li
    className="only-desktop"
  >
    <span
      className="highlight"
    >
      13:15
    </span>
     (
    BOM
    )
  </li>
  <li
    className="only-mobile"
  >
    <span
      className="highlight"
    >
      12:10
       - 
      13:15
    </span>
  </li>
  <li
    className="only-mobile"
  >
    <span>
      DXB
       - 
      BOM
    </span>
  </li>
  <li
    className="only-mobile"
  >
    5 hours 30 minutes
  </li>
  <li>
    <span
      className="color-highlight icon "
    >
      <img
        alt="Flight Emirates (EK512) logo"
        src="test-file-stub"
      />
      Emirates (EK512)
    </span>
  </li>
  <li
    className="only-desktop"
  >
    5 hours 30 minutes
  </li>
  <li>
    <button
      aria-label="Select Flight number Emirates (EK512)"
      aria-pressed={true}
      className="disabled"
      onClick={[Function]}
      title="Select Flight number Emirates (EK512)"
    >
      Select Flight
    </button>
  </li>
</ul>
`;

exports[`Flight Schedule Row renders correctly, no data passed 1`] = `
<ul
  aria-label="Flight undefined departs from undefined at undefined and arrives undefined at undefined, total Flight duration is undefined."
  className="schedule-row "
  role="button"
  tabIndex="0"
>
  <li
    className="only-desktop"
  >
    <span
      className="highlight"
    />
     (
    )
  </li>
  <li
    className="only-desktop"
  >
    <span
      className="highlight"
    />
     (
    )
  </li>
  <li
    className="only-mobile"
  >
    <span
      className="highlight"
    >
       - 
    </span>
  </li>
  <li
    className="only-mobile"
  >
    <span>
       - 
    </span>
  </li>
  <li
    className="only-mobile"
  />
  <li>
    <span
      className="color-highlight icon "
    >
      <img
        alt="Flight undefined logo"
        src="test-file-stub"
      />
    </span>
  </li>
  <li
    className="only-desktop"
  />
  <li>
    <button
      aria-label="Select Flight number undefined"
      className=""
      onClick={[Function]}
      title="Select Flight number undefined"
    >
      Select Flight
    </button>
  </li>
</ul>
`;
